rllib/env/single_agent_episode.py

```diff
@@ -101,8 +102,7 @@ def __init__(
         self.t = self.t_started = (
             t_started if t_started is not None else max(len(self.observations) - 1, 0)
         )
-        if self.t_started < len(self.observations) - 1:
-            self.t = len(self.observations) - 1
+        self._len_pre_buffer = len(self.rewards)
```

Comment:
Added concept of a "lookback buffer" inside an ongoing episode.
This allows for custom connectors to look back at previous data until a certain (user defined) amount of timesteps, e.g. to be able to add "prev. rewards", "prev. 5 actions", etc.. to a model's input (via custom connectors).